reference/constraints/Callback.rst

```diff
@@ -304,11 +304,12 @@ callback method:
 * A closure.
 
 Concrete callbacks receive an :class:`Symfony\\Component\\Validator\\Context\\ExecutionContextInterface`
-instance as only argument.
+instance as the first argument and the :ref:`payload option <payload>` as the second argument.
 
 Static or closure callbacks receive the validated object as the first argument
-and the :class:`Symfony\\Component\\Validator\\Context\\ExecutionContextInterface`
-instance as the second argument.
+, the :class:`Symfony\\Component\\Validator\\Context\\ExecutionContextInterface`
+instance as the second argument and the :ref:`payload option <payload>` as the
```

Comment:
I'm not sure about the `:ref:` I used. When I build locally it's broken: href is `Sequentially.html#payload`

What syntax should I use?

---

It's in `.. include:: /reference/constraints/_payload-option.rst.inc`

![image](https://github.com/symfony/symfony-docs/assets/9253091/a0793516-d2ef-40db-96dc-33c5e3286e94)

Reply:
This is tricky:

* RST creates the anchors based on the section titles
* We're including a RST doc with a section title called `Payload`
* So, the `payload` cross reference exists and it's valid

But, I don't like this automatic references much because if you change the title, then the reference breaks. To avoid this, while merging I've added a explicit reference name in the included file and used it in the links. See https://github.com/symfony/symfony-docs/commit/61fcb40c53d12328d2658747cfa05de4acf1a510

Reply:
> So, the payload cross reference exists and it's valid

This behavior is very strange, and can confuse user because it redirect to another page. `https://symfony.com/doc/current/reference/constraints/Ip.html#reference-constraints-payload` for example

Do you know if we can improve this, in which repository ?

Reply:
Not sure if it's fixable in symfony-tools repository or in the underlying RST parser 😐

Reply:
I created issue https://github.com/doctrine/rst-parser/issues/275 with a reproducer and fix proposal (I hope :smile: )